doc/feature/restAPI.md

```diff
@@ -0,0 +1,71 @@
+Publishing the REST API exposed by the HTTP Server
+============
+
+The http server exposes C++ objects via a REST API, in a JSON serialized format.
+The purpose of this document is to describe how this API request is implemented
+and invoked.
+
+## Requirements
+
+* Define how to invoke the application REST API
+* Implement the JSON representation of the REST API
+
+## API
+
+* _GET /registry_ returns the URIs for all registered objects, together with the
+  supported REST operation (GET or PUT).
+* Each object can be queried for its schema using a GET operation on the URI of
+  that object, suffixed by /schema.
```

Comment:
Imo this would make an awkward REST API. You can still get a /myns/Schema/schema.